File: Week1/Vibhor/Wednesday/Q2ArrayIntersection.java
package Week1.Vibhor.Wednesday;

import java.util.HashSet;


public class Q2ArrayIntersection {
    


    public static void main(String[] args) {

        Solution obj = new Solution();

        int[] nums1 = {1, 2, 2, 1};
        int[] nums2 = {2, 2};

        int[] result = obj.intersection(nums1, nums2);

        for (int num : result) {
            System.out.print(num + " ");
        }
    }



    public int[] intersection(int[] nums1, int[] nums2) {

        HashSet<Integer> set1 = new HashSet<>();
        HashSet<Integer> set2 = new HashSet<>();

        for (int num : nums1) {
            set1.add(num);
        }

        for (int num : nums2) {
            if (set1.contains(num)) {
                set2.add(num);
            }
        }

        int[] ar = new int[set2.size()];
        int i = 0;

        for (int num : set2) {
            ar[i++] = num;
        }

        return ar;
    }
}

File: Week1/Vibhor/Wednesday/q2_array_intersection.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        set1 = set()
        set2 = set()

        for i in range(len(nums1)):
            if(nums1[i] not in set1):
                set1.add(nums1[i])
            
        for i in range(len(nums2)):
            if(nums2[i] in set1):
                set2.add(nums2[i])
        
        ar = [0]*len(set2)
        j=0
        for item in set2:
            ar[j] = item
            j = j+1

        return ar

def main():
    nums1 = [1, 2, 2, 1]
    nums2 = [2, 2]

    obj = Solution()

    result = obj.intersection(nums1, nums2)

    print(result)

if __name__ == "__main__":
    main()